### PR TITLE
Issue #1736: Add ExceptionDiagnoser

### LINKED ISSUE #1736: Consider adding an ExceptionDiagnoser
This would track the frequency of exceptions thrown during the operation, potentially bucketed by exception type.  I expect the diagnoser could use AppContext.FirstChanceException to get the relevant information.

### PR DESCRIPTION
**Pull request by Serg046**
Fixes #1736

Please let me know if you want to add an exception type somehow or add a new changelog file for the next version including this new diagnoser.

### CODE REVIEW

File: src/BenchmarkDotNet/Columns/Column.cs [hunk 1]
```diff
@@ -55,6 +55,7 @@ public static class Column
         public const string CompletedWorkItems = "Completed Work Items";
         public const string LockContentions = "Lock Contentions";
         public const string CodeSize = "Code Size";
+        public const string ExceptionFrequency = "Exception frequency";
```
Comment (adamsitnik) [inline src/BenchmarkDotNet/Columns/Column.cs:58]: I've run following command:

```cmd
cd samples\BenchmarkDotNet.Samples
dotnet run -c Release -f net6.0 --filter *IntroExceptionDiagnoser* --memory
```

And I got:

|                 Method | Exception frequency |   Gen0 | Allocated |
|----------------------- |--------------------:|-------:|----------:|
| ThrowExceptionRandomly |              0.6005 | 0.0229 |     207 B |
|              DoNothing |                   - |      - |         - |

I am wondering whether we should change "Exception frequency" to just "Exceptions" to save some space in the produced output.
```suggestion
        public const string Exceptions = "Exceptions";
```

File: src/BenchmarkDotNet/Diagnosers/ExceptionDiagnoser.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+        public IEnumerable<ValidationError> Validate(ValidationParameters validationParameters) => Enumerable.Empty<ValidationError>();
+
+        private class ExceptionsFrequencyMetricDescriptor : IMetricDescriptor
+        {
+            internal static readonly IMetricDescriptor Instance = new ExceptionsFrequencyMetricDescriptor();
+
+            public string Id => "ExceptionFrequency";
+            public string DisplayName => Column.ExceptionFrequency;
+            public string Legend => "the frequency of exceptions thrown during the operation";
```
Comment (adamsitnik) [inline src/BenchmarkDotNet/Diagnosers/ExceptionDiagnoser.cs:46]:
```suggestion
            public string Legend => "Exceptions thrown per single operation";
```

**Review by adamsitnik — COMMENTED** [review]
Overall, it looks great! I've left some comments, but they are just polishing suggestions.

Thank you for your contribution @Serg046 !

**Comment by Serg046** [comment]
Thanks @adamsitnik, all the notes are good and applied

**Review by adamsitnik — APPROVED** [review]
LGTM, thank you very much @Serg046 !

**Pull request merged**